File: nhglib-core/src/io/github/voidzombie/nhglib/input/models/PointerSourceType.java
package io.github.voidzombie.nhglib.input.models;

/**
 * Created by Fausto Napoli on 01/03/2017.
 */
public enum PointerSourceType {
    POINTER_XY,
    POINTER_DELTA_XY;

    public static PointerSourceType fromString(String value) {
        PointerSourceType res = null;

        switch (value) {
            case "pointerXY":
                res = POINTER_XY;
                break;

            case "pointerDeltaXY":
                res = POINTER_DELTA_XY;
                break;
        }

        return res;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/enums/PointerSourceType.kt
package io.github.movementspeed.nhglib.input.enums

/**
 * Created by Fausto Napoli on 01/03/2017.
 */
enum class PointerSourceType {
    POINTER_XY,
    POINTER_DELTA_XY;

    companion object {

        fun fromString(value: String): PointerSourceType? {
            var res: PointerSourceType? = null

            when (value) {
                "pointerXY" -> res = POINTER_XY
                "pointerDeltaXY" -> res = POINTER_DELTA_XY
            }

            return res
        }
    }
}
